lib/matplotlib/tests/test_animation.py

```diff
@@ -514,5 +514,9 @@ def test_movie_writer_invalid_path(anim):
     else:
         match_str = re.escape("[Errno 2] No such file or directory: '/foo")
     with pytest.raises(FileNotFoundError, match=match_str):
-        _ = anim.save("/foo/bar/aardvark/thiscannotreallyexist.mp4",
-                      writer=animation.FFMpegFileWriter())
+        anim.save("/foo/bar/aardvark/thiscannotreallyexist.mp4",
+                  writer=animation.FFMpegFileWriter())
+    # Prevent the warning on cleanup pointing out that the Animation was not saved.
+    # Note, we can't catch the warning in this function because it happens in __del__
+    # and the `anim` fixture still holds a reference.
+    anim._draw_was_started = True
```

Comment:
Perhaps we could consider to suppress that warning internally on a failed save (i.e. by setting _draw_was_started earlier)?  In my view the user already got a relevant error message; adding a warning from the destructor doesn't really help.

Reply:
I was against this, but in the course of writing out a defense of why still warning in the correct behavior I came to @anntzer 's point of view.

That said, this is a pragmatic fix to the tests and might as well just go in.

Reply:
OK, I have done that then.